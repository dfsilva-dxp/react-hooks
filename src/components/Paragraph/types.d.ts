export interface IParagraphProps {
  text: string;
}
