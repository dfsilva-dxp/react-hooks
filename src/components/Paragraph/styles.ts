import styled, { css } from "styled-components";

export const ParagraphWrapper = styled.p`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
  `}
`;
