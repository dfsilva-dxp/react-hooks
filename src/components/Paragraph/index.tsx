import * as S from "./styles";

import { IParagraphProps } from "./types";

export default function Paragraph({ children }: IParagraphProps) {
  return <S.ParagraphWrapper>{children}</S.ParagraphWrapper>;
}
