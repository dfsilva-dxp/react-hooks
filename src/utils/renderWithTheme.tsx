import { ReactNode } from "react";
import { render, RenderResult } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { MyTheme } from "../styles";

export const renderWithTheme = (children: ReactNode): RenderResult =>
  render(<ThemeProvider theme={MyTheme}>{children}</ThemeProvider>);
