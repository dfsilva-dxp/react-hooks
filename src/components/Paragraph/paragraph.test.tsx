import { screen } from "@testing-library/dom";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Paragraph from ".";
describe("<Paragraph/>", () => {
  it("should render a title", () => {
    const { container } = renderWithTheme(
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.{" "}
      </Paragraph>
    );

    expect(
      screen.getByText(
        /Lorem ipsum dolor sit amet, consectetur adipisicing elit./i
      )
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
