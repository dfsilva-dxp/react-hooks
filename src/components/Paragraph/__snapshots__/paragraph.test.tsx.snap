// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph/> should render a title 1`] = `
.c0 {
  margin-top: 2rem;
}

<div>
  <p
    class="c0"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit.
     
  </p>
</div>
`;
